internal/shim-sgx/src/start.rs

```diff
@@ -0,0 +1,295 @@
+//! This is the elf entry point called by the hypervisor
+
+//! enclave enter
+//!
+//! see [`_start`](_internal::_start)
+
+mod _internal {
+    use const_default::ConstDefault;
+    use syscall::SYS_ENARX_ERESUME;
+    use xsave::XSave;
+
+    const GPR: u64 = 4096 - 184;
+    const RSPO: u64 = GPR + 32;
+    const MISC: u64 = GPR - 16;
+    const SRSP: u64 = MISC - 8;
+
+    const STACK: u64 = 9 * 8;
+    const SHIM: u64 = 10 * 8;
```

Comment:
@lkatalin @npmccallum maybe you can document these and make suggestions how to create this computationally.